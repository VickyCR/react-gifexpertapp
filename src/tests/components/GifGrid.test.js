import React from 'react';
import { shallow } from 'enzyme';

import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('GifGrid component', () => {

  test('should create', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category="Goku" />);

    expect(wrapper).toMatchSnapshot();
  });

  test('should display the items from useFetchGifs', () => {
    const gifs = [{
      id: '313221313',
      title: 'Title Image',
      url: 'http://domain.com/image.png',
    }];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category="Goku" />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBeFalsy();
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
