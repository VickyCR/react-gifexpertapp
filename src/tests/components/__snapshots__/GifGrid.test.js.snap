// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid component should create 1`] = `
<Fragment>
  <h3>
    Goku
  </h3>
  <p
    className="animate__animated animate__flash animate__infinite"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`GifGrid component should display the items from useFetchGifs 1`] = `
<Fragment>
  <h3>
    Goku
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="313221313"
      key="313221313"
      title="Title Image"
      url="http://domain.com/image.png"
    />
  </div>
</Fragment>
`;
